code/game/turfs/turf.dm

```diff
@@ -732,6 +735,28 @@
 
 	return protection_level
 
+/turf/proc/get_aa_with_highest_protection_level(faction)
+	var/obj/structure/machinery/defenses/planetary_anti_air/highest_aa = null
+	var/protection_level = TURF_PROTECTION_NONE
+
+	for (var/obj/structure/machinery/defenses/planetary_anti_air/planetary_aa in src.linked_aa)
```

Comment:
```suggestion
	for (var/obj/structure/machinery/defenses/planetary_anti_air/planetary_aa in linked_aa)
```
only need to do src.var in initialize proc, do that to every var that has src.var in this pr